API/api/controllers/event_controller.py

```
@@ -19,4 +19,13 @@ def create_event(request):
     except ValidationError as e:
-        return Response(e.detail, status=status.HTTP_400_BAD_REQUEST)
\ No newline at end of file
+        return Response(e.detail, status=status.HTTP_400_BAD_REQUEST)
+    
+@api_view(["GET"])
+@permission_classes([IsAuthenticated])
+def get_upcoming_events(request):
+    try:
+        events = service.get_upcoming_events()
+        return Response(events, status=status.HTTP_200_OK)
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)